
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Playing with microsatellites (Simple Sequence Repeats), Java, and Neo4j - oh no sequences!</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="Hi! I just finished this afternoon a small project I had to do about identification of microsatellites in DNA sequences. As with every new project I &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ohnosequences.com/blog/2011/10/18/playing-with-microsatellites-simple-sequence-repeats-java-and-neo4j">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="oh no sequences!" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-4']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1>
    <a href="/">
      <span class="title_oh">oh</span><span class="title_no">no</span><span class="title_sequences">sequences</span><span class="title_bang">!</span>
    </a>
  </h1>
  <h2>era7 bioinformatics R&D group</h2>

</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:ohnosequences.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/publications">publications</a></li>
  <li><a href="/projects">projects</a></li>
  <li><a href="/people">people</a></li>
  <li><a href="/about">about</a></li>
  <li><a href="/contact">contact</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      <h1 class="entry-title">Playing with microsatellites (Simple Sequence Repeats), Java, and Neo4j</h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-10-18T21:28:04+02:00" pubdate data-updated="true">Oct 18<span>th</span>, 2011</time>
        
      </p>
    
  </header>


<div class="entry-content"><p>Hi!</p>

<p>I just finished this afternoon a small project I had to do about identification of microsatellites in DNA sequences. As with every new project I start, I think of something that:</p>

<ul>
  <li>I <strong>didn’t try before</strong></li>
  <li>is <strong>worth learning</strong></li>
  <li>is <strong>applicable</strong> in order to meet the needs of the specific project</li>
</ul>

<p>These last few days it was the chance to get to know and try the visualization tool included in the last version of <a href="http://blog.neo4j.org/2011/10/announcing-neo4j-15-boden-bord.html"><strong>Neo4j Webadmin dashboard</strong></a>.
I had already heard of it a couple of times from different sources but had not had the chance to play a bit with it yet. So, after my first contact with it I have to say that although it’s something <a href="http://www.neo4j.org">Neo4j</a> introduced in the last versions, it already has a decent GUI and promising functionality. </p>

<p>Apart from GUI considerations, I created the repository <a href="https://github.com/pablopareja/MicrosatellitesNeo4jModel"><strong>MicrosatellitesNeo4jModel</strong></a> with a bunch of nodes and relationships wrappers as an API for  performing traversals for all this data in an easy way.
Here is the domain model I chose:</p>

<p><a href="https://s3-eu-west-1.amazonaws.com/pablo-tests/MicrosatellitesDomainModel.svg"><img src="https://s3-eu-west-1.amazonaws.com/pablo-tests/MicrosatellitesDomainModel.jpg" alt="Microsatellites Neo4j domain model" /></a></p>

<p>On the programs side, I developed two different Java classes, one dealing with the identification of the microsatellites and their subsequent storage on the Neo4j DB (<a href="https://github.com/pablopareja/Microsatellites/blob/develop/src/com/era7/bioinfo/microsats/programs/CreateMicrosatellitesDB.java">CreateMicrosatellitesDB.java</a>) and another (<a href="https://github.com/pablopareja/Microsatellites/blob/develop/src/com/era7/bioinfo/microsats/programs/ExtractDataToCSV.java">ExtractDataToCSV.java</a>)for extracting statistical information for a set of specific parameters like tuple length and things like that. 
Both classes are in the repository <a href="https://github.com/pablopareja/Microsatellites"><strong>Microsatellites</strong></a>.</p>

<p>Once the DB was created, I played a bit with the display profiles in the WebAdmin data browser so that different node types had a different aspect and this is what I got:</p>

<p><a href="https://s3-eu-west-1.amazonaws.com/pablo-tests/MicrosatellitesDataBrowserScreenshot.jpg"><img src="https://s3-eu-west-1.amazonaws.com/pablo-tests/MicrosatellitesDataBrowserScreenshot.jpg" alt="Microsatellites DB data browser screenshot" /></a></p>

<p>Here you can find blue circles (<strong>sequence IDs</strong>), orange boxes (<strong>tuples</strong> repeated in the microsatellites found), and greenish squares (<strong>tuple length</strong> nodes). </p>

<p>One of the features I was missing in the visualization was having style rules for relationships as well as for nodes. This was specially important in my case where I have relevant information stored as relationships attributes, <em>(I actually could not visualize the number of tuple repeats in the microsatellites found, just the name of the relationship ‘MICROSATELLITE_FOUND’ everywhere)</em>.
However I posted a question on neo4j user list about this and it seems they already are working on this, cool!</p>

<p>As always, everything here is open source and released under under <a href="http://www.gnu.org/licenses/agpl.html"><strong>AGPLv3</strong></a>.</p>

<p>Cheers,</p>

<p><a href="http://twitter.com/pablopareja">@pablopareja</a></p>

</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Pablo Pareja</span></span>

      








  


<time datetime="2011-10-18T21:28:04+02:00" pubdate data-updated="true">Oct 18<span>th</span>, 2011</time>
      

<span class="categories">
  
    <a class='category' href='/blog/categories/neo4j/'>neo4j</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://ohnosequences.com/blog/2011/10/18/playing-with-microsatellites-simple-sequence-repeats-java-and-neo4j/" data-via="ohnoseq" data-counturl="http://ohnosequences.com/blog/2011/10/18/playing-with-microsatellites-simple-sequence-repeats-java-and-neo4j/" >Tweet</a>
  
  
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/blog/2011/10/14/some-thoughts-about-neo4j/" title="Previous Post: Some thoughts about Neo4j">&laquo; Some thoughts about Neo4j</a>
      
      
        <a class="basic-alignment right" href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/" title="Next Post: Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly &raquo;</a>
      
    </p>
  </footer>
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2015/03/22/bio4j-preprint/">Bio4j preprint</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/12/03/intercrossing-cloud-and-ngs-course/">INTERCROSSING cloud and NGS course</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/21/a-short-report-about-the-conference-exploring-human-host-microbiome-interactions-in-health-and-disease-at-wellcome-trust-hinxton-on-may-8-10-2012/">A short report about the Conference Exploring Human Host-Microbiome Interactions in Health and Disease at Wellcome Trust, Hinxton on May 8 - 10, 2012 </a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/23/neo4j-server-and-aws-become-good-friends/">Neo4j Server and AWS become good friends!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/ohnosequences">@ohnosequences</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'ohnosequences',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
