
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Bio4j: a graph based DB containing most of data available in Uniprot, Uniref and Gene Ontology - oh no sequences!</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="We have recently launched Bio4j, a graph based database that includes most of the data available in Uniprot, Uniref clusters and Gene Ontology. &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ohnosequences.com/blog/2011/03/23/bio4j-a-graph-based-db-containing-most-of-data-available-in-uniprot-uniref-and-gene-ontology">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="oh no sequences!" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-4']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1>
    <a href="/">
      <span class="title_oh">oh</span><span class="title_no">no</span><span class="title_sequences">sequences</span><span class="title_bang">!</span>
    </a>
  </h1>
  <h2>era7 bioinformatics R&D group</h2>

</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:ohnosequences.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/publications">publications</a></li>
  <li><a href="/projects">projects</a></li>
  <li><a href="/people">people</a></li>
  <li><a href="/about">about</a></li>
  <li><a href="/contact">contact</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      <h1 class="entry-title">Bio4j: a graph based DB containing most of data available in Uniprot, Uniref and Gene Ontology</h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-03-23T14:29:56+01:00" pubdate data-updated="true">Mar 23<span>rd</span>, 2011</time>
        
      </p>
    
  </header>


<div class="entry-content"><p>We have recently launched Bio4j, a graph based database that includes most of the data available in Uniprot, Uniref clusters and Gene Ontology. Quoting from the Bio4j wiki:</p>

<p> 
&gt; Bio4j provides a completely new and powerful framework for protein related information querying and management. Since it relies on a high-performance graph engine (Neo4j), data is stored in a way that semantically represents its own structure. On the contrary, traditional relational databases must flatten the data they represent into tables, creating “artificial” ids in order to connect the different tuples; which can in some cases eventually lead to domain models that have almost nothing to do with the actual structure of data.</p>

<p>Thanks to its graph structure queries that could be time consuming or even inpossible in relational DBs can be performed easily in Bio4j.</p>

<p>An example of this kind of queries would be the following:</p>

<p>Suppose you are working in the membrane proteins of a non model organism that unfortunately have very poor annotations. Searching for proteins of your interest in Uniprot by your organism name or taxonomy only results in a few unrewieved entries with few more than the sequence. Next step could probably be to use the Uniprot advance search tool to get the proteins similar to your proteins in a 50, 90 or 100% (the Uniref clusters) and then look protein by protein which of them have annotations that could indicate that they are membrane proteins and select those that 1) have good enough annotaions for your purpose and 2) are related enough to your protein to be able to infer the functions of the well-annotated proteins to your protein. This would be a manual and tedious process time consuming and error prone.</p>

<p>All this process of getting all the proteins of your organism, then getting the high similar ones and then getting only those which are in the membrane and have quality annotations could be perfomed programmatically in a fairly simple graph traversal. All you would have to specify would be:</p>

<ul>
  <li>The level of needed similarity between the proteins of your organism and the well-characterized ones</li>
  <li>The criteria to consider a protein has good enough annotations: if the entry is reviewed, if it’s got lots of cross references, of functional data (Interpro, GO…)</li>
  <li>The criteria to consider a protein is in the membrane, for example that it’s got a GO ID indicating that is a membrane protein, that its subcellular location is the membrane… both…</li>
</ul>

<p>More information about Bio4j in its <a href="http://www.bio4j.com">website</a>, the <a href="http://blog.bio4j.com/">blog</a>, the <a href="http://wiki.bio4j.com/">wiki </a> and the <a href="http://twitter.com/#!/bio4j">twitter profile</a>.</p>
</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">oh no sequences!</span></span>

      








  


<time datetime="2011-03-23T14:29:56+01:00" pubdate data-updated="true">Mar 23<span>rd</span>, 2011</time>
      

<span class="categories">
  
    <a class='category' href='/blog/categories/bio4j/'>bio4j</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://ohnosequences.com/blog/2011/03/23/bio4j-a-graph-based-db-containing-most-of-data-available-in-uniprot-uniref-and-gene-ontology/" data-via="ohnoseq" data-counturl="http://ohnosequences.com/blog/2011/03/23/bio4j-a-graph-based-db-containing-most-of-data-available-in-uniprot-uniref-and-gene-ontology/" >Tweet</a>
  
  
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/blog/2011/02/25/using-uniprot-advanced-search/" title="Previous Post: Using Uniprot Advanced search">&laquo; Using Uniprot Advanced search</a>
      
      
        <a class="basic-alignment right" href="/blog/2011/03/25/using-bio4j-go-tools-to-get-the-go-annotations-of-a-set-of-uniprot-proteins-2/" title="Next Post: Using Bio4j Go Tools to get the GO annotations of a set of Uniprot proteins">Using Bio4j Go Tools to get the GO annotations of a set of Uniprot proteins &raquo;</a>
      
    </p>
  </footer>
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2015/03/22/bio4j-preprint/">Bio4j preprint</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/12/03/intercrossing-cloud-and-ngs-course/">INTERCROSSING cloud and NGS course</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/21/a-short-report-about-the-conference-exploring-human-host-microbiome-interactions-in-health-and-disease-at-wellcome-trust-hinxton-on-may-8-10-2012/">A short report about the Conference Exploring Human Host-Microbiome Interactions in Health and Disease at Wellcome Trust, Hinxton on May 8 - 10, 2012 </a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/23/neo4j-server-and-aws-become-good-friends/">Neo4j Server and AWS become good friends!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/ohnosequences">@ohnosequences</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'ohnosequences',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
