---
author: Marina Manrique
comments: false
date: 2011-06-05 20:01:46
layout: post
slug: talks-genomes-reads-and-annotations
title: Talks, Genomes, Reads and Annotations
wordpress_id: 314
categories:
- ngs
tag:
- EHEC
---

Last 1st of June I was attending to the conference "[Applied Bioinformatics & Public Health Microbiology](https://registration.hinxton.wellcome.ac.uk/display_info.asp?id=227)" held at the Welcome Trust Conference Centre (Hinxton, UK)

The main purpose of the conference was to point out the importance of high-throughput technologies (mainly NGS) and bioinformatics in solving issues related to public health.

During the morning sessions of the second day (2nd of June) BGI announced the release of the sequencing data of 5 IonTorrent chips (see the  announcement [here ](http://www.bgisequence.com/eu/index.php?cID=194&sms_ss=twitter&at_xt=4de74702e845e3a2,0)). Just some hours later Nick Loman (@pathogenomenick) published a de novo assembly of the reads with MIRA in his blog (see post [here](http://pathogenomics.bham.ac.uk/blog/2011/06/ehec-genome-assembly/)). And then, some hours later (in the morning of the 3rd of June) we published the annotation of the Nick's assembly in [our website](http://www.era7bioinformatics.com/en/E_Coli_EHEC_O104_STRAIN_EU_OUTBREAK_era7bioinformatics.html). We annotated it with the pipeline (BG7 pipeline) we were presenting at the conference (see talk slides [here](http://www.slideshare.net/marina_manrique/bg7-a-new-system-for-bacterial-genome-annotation-designed-for-ngs-data)).

Now we are working on this automatic annotation data searching for proteins that may be involved in pathogenesis. These preliminary results look promising.

Definitely, it's been a really exciting experience. In the middle of a conference where we were trying to see how NGS could be applied to solve public health issues we could get the genome of the bacteria responsible for this important outbreak. In less than 24 hours we got the reads, the assembly and the annotation. A good case study :)
