---
comments: false
author: ??
date: 2011-02-23 13:07:40
layout: post
slug: bacterial-diversity-identification-with-normalized-libraries
title: Bacterial diversity identification with normalized libraries?
wordpress_id: 130
categories:
- ngs
tag:
- 16S
- ngs
---

Normally the bacterial identification experiments with NGS are based on analyzing 16S rRNA amplicons. Typically with 454 technology. In this kind of experiments the diversity of the bacterial populations found in the samples are infered from the similarity of the reads with already known 16s rRNA sequences. And the abundance of detected species could be measured with the abundance of detected reads of such specie in the sample.

This kind of approach where the abundance of a certain thing is meassured with the abundance of annotated sequences with that thing is also applied in RNA-seq experiments (where the gene expression is such thing) with no normalized libraries.

If the goal of the transcriptome experiment is not to quantify gene expression but detect as many different transcripts as possible then the library is normalized.

I wonder if in the bacterial identificacion experiments using 16s amplicons this approach could also be used in the cases where you want to get the maximum number of species present in your sample.
