<!DOCTYPE html>
<head>
<meta charset="utf-8">
  <meta name="author" content="Pablo Pareja" />
  <title>Bio4j</title>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,700,400italic|Open+Sans:300italic,400italic,600italic,400,300,600' rel='stylesheet' type='text/css'>
<style>
  html { background-color: #425331; }
  body { 
    background-color: #f9fcfe; 
    color: #414141;
    text-align: left;
  }
  /* A section is a slide. It's size is 800x600, and this will never change */
  section {

    display: inline-block;
    font-family: Open Sans, sans;
    font-weight: 300;
    font-size: 225%; /* 16 * 2.25 = 36 */
    color: #535353;
  }

  p { margin-bottom: 1.333em; }
  strong { color: #b8913d; } 
  em {  font-weight: 400; color: #121212; }
  code {  font-family: PT Mono; font-size: 76%; }
  address, blockquote, dl, fieldset, p, form, h1, h2, h3, h4, 
  h5, h6, hr, ol, pre, table, ul, dl { 
    vertical-align: middle;
  }
  h1, h2, h3 {
    margin-top:0; margin-bottom:0;
    letter-spacing: -1px;
    text-align: center;
    font-family: Bitter;
    font-weight: normal;
    color: #b8463d;
  }
  h1 {  font-size:2em; padding: 0.6666em 0; }
  h1 { padding-top: 0; }
  h2 {  font-size:1.75em; padding:0.381em 0;  }
  h3 {  
    font-size: 1.5em; 
    padding-top: 0;
    padding-bottom: 0.8888em;
    padding-left: 0;
    padding-right: 0;
    text-align: left; 
  }
  section.titleslide h1 {}
  h1.title { 
    text-align: right;
    font-size: 2.4em;
    font-family: Bitter;
    font-weight: bold; 
    font-style: italic;
    /*color: #b8913d;*/
    color: #b8463d;
    padding: 0.6666em 0;
    padding-bottom: 0;
  }
    h2.author { 
    font-size: 1.25em;
    font-family: Open Sans;
    text-align: right;
    margin: 0 0 0 0;
    font-weight: 300; color:#266871; 
  }
  h3.date { 
    text-align: right;
    font-size: 1em;
    font-family: Open Sans;
    margin: 0 0 0 0;
    font-weight: 300; font-style: normal;
  }
  ul { list-style-type: round;}
  ol { list-style-type: decimal;}
  ul, ol {
    margin: 10px 10px 10px 80px;
  }
  li { text-align: left; }
  q { quotes: "“" "”" "‘" "’"; }
  blockquote { 
    margin-top: 1.333em;
    padding-top: 1.333em;
    padding-bottom: 1.333em; 
    padding-left: 2em;
    padding-right: 2em;
    font-style: italic;
    text-align: right;
    border-right: 6px solid #b8463d;
    background-color: #f2f2f2;
  }
  a:link {color:#266871; text-decoration: none;}      /* unvisited link */
  a:visited {color:#266871; text-decoration: none;}  /* visited link */
  a:hover {color:#569ca6;}  /* mouse over link */
  a:active {color:#3e7f89;}  /* selected link */

  /* Figures are displayed full-page, with the caption on
     top of the image/video */
  figure {
    background-color: transparent;
    overflow: visible;
  }
  img:first-of-type {
    margin-left: -80px;
    margin-top: -120px;
    position:absolute;
    width: 800px;
    height: 600px;
    top: 0;
    left: 0;
    z-index: -1;
  }
  figure > img {
    width: 800px !important;
    height: 600px !important;
    margin-left: -80px;
    margin-top: -120px;
  }
  figcaption {
    margin: 70px;
  }
  footer {
    position: absolute;
    bottom: 0;
    width: 100%;
    padding: 40px;
    text-align: right;
    background-color: #F3F4F8;
    border-top: 1px solid #CCC;
  }

  /* Transition effect */
  /* Feel free to change the transition effect for original
     animations. See here:
     https://developer.mozilla.org/en/CSS/CSS_transitions
     How to use CSS3 Transitions: */
  section {
      -moz-transition: left 200ms ease-out 0s;
      -webkit-transition: left 200ms ease-out 0s;
      -ms-transition: left 200ms ease-out 0s;
      transition: left 200ms ease-out 0s;
  }

  /* Before */
  section { 

    width: 640px;
    position: absolute;
    height:360px;
    top: 50%; left: 50%;
    margin-left: -320px; margin-top: -180px;
    vertical-align: middle;
    left: -150%; }
  /* Now */
  section[aria-selected] { 
    /* 16:9 640x360 */
    width: 640px;
    position: absolute;
    height:360px;
    top: 50%; left: 50%;
    margin-left: -320px; margin-top: -180px;
    vertical-align: middle;
    display: inline-block;
  }
  /* After */
  section[aria-selected] ~ section { 

    width: 640px;
    position: absolute;
    height:360px;
    top: 50%; left: 50%;
    margin-left: -320px; margin-top: -180px;
    vertical-align: middle;
    left: +150%; 
  }

  /* Incremental elements */

  /* By default, visible */
  .incremental > * { opacity: 1; }

  /* The current item */
  .incremental > *[aria-selected] { color: #b8913d; opacity: 1; font-weight: 500;}

  /* The items to-be-selected */
  .incremental > *[aria-selected] ~ * { opacity: 0.3; }

  /*logos*/
  #logos_div {
    position: absolute;
    
  }
  #ohnoseq_logo_div {}
  #ohnoseq_logo {
    /*6.132867133*/
    height: 20px; width: 123px;
    top:570px;
    left: 730px;
    position: relative;
  }
  #era7_logo_div {}
  #era7_logo {
    /*4.866071429*/
    height: 20px; width: 97px;
    position: relative;
    top: 540px;
    left: 755px;
  }
  #bio4j_logo {
    /*1.236979167*/
    height: 64px; width: 80px;
    position: relative;  
    top:45px;
    left: 780px;
  }
</style>
</head>
<body>
<section>
  <h1 class="title">Bio4j</h1>
  <h2 class="author"><strong><a href="http://twitter.com/pablopareja">Pablo Pareja</a></strong></h2>
  <h3 class="date"><a href="http://graphdevroom.org"><strong>GraphDevRoom 2014</strong></a></h3>
</section>
<div id="logos_div">
  <div id="ohnoseq_logo_div">
    <a href="http://ohnosequences.com">
      <img id="ohnoseq_logo" src="resources/logos/ohnoseq-logo.png">
    </a>
  </div>
  </div>
  <div id="era7_logo_div">
    <a href="http://era7bioinformatics.com">
      <img id="era7_logo" src="resources/logos/era7-logo.png">
    </a>
  </div>
  <div id="bio4j_logo_div">
    <a href="http://bio4j.com">   
      <img id="bio4j_logo" src="resources/logos/bio4j-logo.jpg">
    </a>
  </div>
</div> 
<section class="slide level2">

<!-- 
  Sections

  - what is Bio4j
  - why Bio4j
      - why graphs bio stuff
      - why graph DBs
      - why cloud
  - details?
      - data
      - model
      - technologies
      - other stuff -> dev etc
  - how Bio4j
      - use cases
      - case studies
      - contact/support/dev/whatever
  - who Bio4j
      - era7 and oh no sequences!
      - team
-->

<p><br/> <br/></p>
<h3 id="what-is-bio4j"><em>what</em> is Bio4j</h3>
</section>
<section class="slide level2">

<h3 id="in-one-sentence">in one sentence</h3>
<p><strong>Bio4j</strong> is a bioinformatics <em>graph</em>-based data platform <strong>integrating</strong> most data available in the most representative <strong>open data sources</strong> around <strong>protein information</strong> available today.</p>
</section>
<section class="slide level2">

<h3 id="data">data</h3>
<ul>
<li><em><a href="http://www.uniprot.org/help/uniprotkb">UniProt KB</a></em> (SwissProt + Trembl)</li>
<li><em><a href="http://www.geneontology.org/">Gene Ontology</a></em> (GO)</li>
<li><em><a href="http://www.uniprot.org/help/uniref">UniRef</a></em> (50,90,100)</li>
<li><em><a href="http://www.ncbi.nlm.nih.gov/RefSeq/">RefSeq</a></em></li>
<li><em><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/">NCBI taxonomy</a></em></li>
<li><em><a href="http://enzyme.expasy.org/">Expasy Enzyme DB</a></em></li>
</ul>
<!-- 4 or more '-' create a new slide -->

</section>
<section class="slide level2">

<h3 id="open">open!</h3>
<ul>
<li><strong>code</strong> <strong><a href="https://www.gnu.org/licenses/agpl-3.0.html">AGPLv3</a></strong></li>
<li><strong>data</strong> integrates only <strong><a href="http://okfn.org/opendata/">open data</a></strong></li>
<li><strong>implementation &amp; release</strong> process is 100% public and totally <strong>transparent</strong></li>
</ul>
</section>
<section class="slide level2">

<p><br/> <br/></p>
<h3 id="why-bio4j"><em>why</em> Bio4j?</h3>
<p>bio data + graph databases + the cloud</p>
</section>
<section class="slide level2">

<h3 id="biology-dbs-today">biology &amp; DBs today</h3>
<p>Highly <strong>interconnected</strong> overlapping knowledge <strong>spread</strong> through <em>different databases</em></p>
</section>
<section class="slide level2">

<figure>
<img src="resources/images/overlappingDBsBiology.jpg" />
</figure>
</section>
<section class="slide level2">

<h3 id="why-graphs">why graphs</h3>
<p>In most cases all data is modeled in <em><strong>Relational Databases</strong></em> or sometimes even just as plain <em><strong><code>CSV</code> files</strong></em>.</p>
</section>
<section class="slide level2">

<h3 id="why-graphs-1">why graphs</h3>
<p>That might be OK for simple scenarios but as the <strong>amount</strong> and <strong>diversity</strong> of <strong>data grows</strong>, <strong>domain models</strong> become crazily <strong>complicated</strong>!</p>
</section>
<section class="slide level2">

<figure>
<img src="resources/images/GOdomainModel.jpg" alt="Doesn’t look very compelling right? :)" /><figcaption><em>Doesn’t look very compelling right? :)</em></figcaption>
</figure>
</section>
<section class="slide level2">

<h3 id="why-graphs-2">why graphs</h3>
<p>With a relational paradigm the double implication</p>
<p><strong>Entity</strong> &lt;–&gt; <strong>Table</strong></p>
<p>does not go both ways</p>
</section>
<section class="slide level2">

<h3 id="not-so-good-implications"><em>not-so-good</em> implications</h3>
<ul>
<li><strong>Auxiliary tables</strong></li>
<li><strong>Artificial IDs</strong></li>
<li>Dealing with <strong>raw tables</strong> (in spite of Entity-relationship diagrams)</li>
<li><strong>Integrating</strong> new knowledge becomes <strong>difficult</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="biology-table">biology <em>!=</em> table</h3>
<p><strong>Life</strong> in general and <strong>biology</strong> in particular are probably not 100% like a graph…</p>
<p>but one thing’s sure, they <em><strong>are not a set of tables!</strong></em></p>
</section>
<section class="slide level2">

<h3 id="why-graph-databases">why graph databases</h3>
<ul>
<li><strong>Data</strong> stored in a way that <strong>semantically represents its own structure</strong></li>
<li>Incorporating new data is easy –&gt; <strong>scalability</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="why-graph-databases-1">why graph databases</h3>
<ul>
<li><strong>Vertex-centric</strong> <em>(local)</em> indices allow for complex traversals –&gt; overcoming <strong>supernode problem</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="cloud">cloud</h3>
<ul>
<li>data as a service</li>
<li>machine configurations</li>
</ul>
</section>
<section class="slide level2">

<p><br/> <br/></p>
<h3 id="details-about-bio4j"><em>details</em> about Bio4j</h3>
<p>data, model, technologies, APIs…</p>
</section>
<section class="slide level2">

<!-- Details about Bio4j -->

<p><br/> <br/></p>
<h3 id="a-bit-of-history">a bit of <em>history</em></h3>
<p>From the beginnings to the BigData platform it is today</p>
</section>
<section class="slide level2">

<h3 id="how-it-all-started">How it all started</h3>
<ul>
<li>Need for <strong>massive access</strong> to <em>Gene Ontology</em> annotations</li>
<li><a href="http://bg7.ohnosequences.com/"><strong>BG7</strong></a> bacterial genome annotation system</li>
<li>Need for massive direct access to <strong>protein information</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="more-and-more-data">more and more data!</h3>
<ul>
<li>As <em>other</em> data sources were becoming a <em>bottleneck</em> they were being added to Bio4j</li>
<li>First it was Uniprot KB, then Uniref and <strong>we didn’t stop yet :)</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="numbers">numbers</h3>
<ul>
<li><span class="math">10<sup>9</sup></span> edges</li>
<li><span class="math">2 × 10<sup>8</sup></span> nodes</li>
<li><span class="math">6 × 10<sup>8</sup></span> properties</li>
<li><span class="math">150</span> edge types</li>
<li><span class="math">40</span> node types</li>
</ul>
<!-- * **`1,216,993,547`** relationships
* `190,625,351` nodes
* `584,436,429` properties
* `144` relationship types
* `42` node types
 -->
</section>
<section class="slide level2">

<h3 id="bio4j-structure">Bio4j structure</h3>
<p>Bio4j importing process is <strong>modular</strong> and <strong>customizable</strong> allowing you to import just the data you are interested in.</p>
</section>
<section class="slide level2">

<h3 id="data-sources---modules-i">data sources - modules I</h3>
<ul class="incremental">
<li>Gene Ontology (GO)</li>
<li>ExPASy Enzyme DB</li>
<li>RefSeq</li>
</ul>
</section>
<section class="slide level2">

<h3 id="data-sources---modules-ii">data sources - modules II</h3>
<ul class="incremental">
<li>UniRef -&gt; 50, 90, 100</li>
<li>NCBI taxonomy tree -&gt; GI index</li>
<li>Uniprot KB -&gt; Swissprot/Trembl, interactions…</li>
</ul>
</section>
<section class="slide level2">

<h3 id="data-sources---modules-iii">data sources - modules III</h3>
<p>Just keep in mind that you must be <strong>coherent</strong></p>
<p><em>e.g. you cannot import protein interactions if you didn’t import any protein yet!</em></p>
</section>
<section class="slide level2">

<h3 id="bio4j-apis">Bio4j APIs</h3>
<ol class="incremental" type="1">
<li>abstract <strong>domain model</strong></li>
<li><strong>Blueprints</strong> implementation</li>
<li><strong>technology-specific</strong> versions</li>
</ol>
</section>
<section class="slide level2">

<h3 id="domain-model">domain model</h3>
<p>Bio4j database has a <strong>well-defined</strong> domain model and all nodes and relationships comply with this abstract model</p>
</section>
<section class="slide level2">

<figure>
<img src="resources/images/Bio4jDomainModelWithCardinality.jpg" />
</figure>
</section>
<section class="slide level2">

<h3 id="domain-model-why">domain model <em>why?</em></h3>
<ul>
<li>abstract over Blueprints</li>
<li>more precise <strong>typing</strong></li>
<li>implementations can use technology-specific features</li>
</ul>
</section>
<section class="slide level2">

<h3 id="key-advantage">Key advantage</h3>
<p><em>Different</em> <strong>graph topologies</strong> at the storage level, <em>same</em> <strong>domain model</strong>.</p>
<p>Example: use <strong>type nodes</strong> in <em>Titan</em>, <strong>labels</strong> in <em>Neo4j</em>.</p>
</section>
<section class="slide level2">

<h3 id="blueprints-layer">Blueprints layer</h3>
<p>A default <strong><a href="https://github.com/tinkerpop/blueprints/">Blueprints</a></strong> implementation of the abstract model.</p>
</section>
<section class="slide level2">

<p>Apart from the set of interfaces developed as the <strong>first layer</strong> for the <em>domain model</em> there’s an <strong>extra layer</strong> that uses <em>Blueprints</em>. This way we’re going one step further for making the domain model <strong>independent</strong> from the choice of <em>database technology</em></p>
</section>
<section class="slide level2">

<h3 id="technology-specific">technology-specific</h3>
<p>Optimizations, features, etc.</p>
<ul>
<li><strong><a href="https://github.com/neo4j/neo4j">Neo4j</a></strong></li>
<li><strong><a href="https://github.com/thinkaurelius/titan/">Titan</a></strong> (WIP)</li>
<li><strong><a href="https://github.com/orientechnologies/orientdb/">OrientDB</a></strong> (planned)</li>
</ul>
</section>
<section class="slide level2">

<h3 id="why-neo4j">why Neo4j</h3>
<ul class="incremental">
<li>wide <strong>adoption</strong></li>
<li>stable</li>
<li><strong>Cypher</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="why-titan">why Titan</h3>
<ul class="incremental">
<li><strong>local!</strong> indexes</li>
<li><strong>on-disk</strong> access</li>
<li><strong>type</strong> definitions -&gt; <em>constraints!</em></li>
</ul>
</section>
<section class="slide level2">

<h3 id="bio4j-and-the-cloud">Bio4j and the cloud</h3>
<ul>
<li><strong>Interoperability</strong> and data distribution</li>
<li><strong>Backup</strong> and <strong>storage</strong></li>
<li><strong>Scalability</strong></li>
<li>Applications and service providers on the cloud</li>
<li>Cost-effective</li>
</ul>
</section>
<section class="slide level2">

<h3 id="dev-and-release-process">dev and release process</h3>
<ul>
<li>coordinate <strong>data</strong> and <strong>code</strong></li>
<li><strong><a href="http://semver.org/spec/v2.0.0.html">Semantic Versioning</a></strong></li>
<li><strong>Cloud</strong> integration, distribution, deployment, …</li>
</ul>
</section>
<section class="slide level2">

<h3 id="how">how?</h3>
<ul>
<li><strong><a href="http://ohnosequences/statika">Statika</a></strong> cloud, data + code, modules (see <a href="https://fosdem.org/2014/schedule/event/graphdevroom_bio4j_1/">next talk</a>)</li>
<li><strong><a href="https://github.com/sbt/sbt">sbt</a></strong> build Java + Scala, automated Bio4j-specific test &amp; release</li>
<li><strong><a href="https://github.com/bio4j">git + github</a></strong> versioning, docs, collaboration, coordination</li>
</ul>
</section>
<section class="slide level2">

<p><br/> <br/></p>
<!-- How Bio4j -->
<h3 id="how-to-use-bio4j"><em>how</em> to use Bio4j?</h3>
<p>use cases, case studies, community</p>
</section>
<section class="slide level2">

<h3 id="use-cases">use cases</h3>
</section>
<section id="section" class="slide level2">
<h1><!--
----

### ??? and !!!

I don't get what's this... :|
--></h1>
<h3 id="how-we-use-it">how we use it</h3>
<ul>
<li><strong><a href="http://bg7.ohnosequences.com">bg7</a></strong> genome annotation</li>
<li><strong>mg7</strong> metagenomics analysis</li>
<li>comparative genomics, network analysis, genome assembly, …</li>
</ul>
</section>
<section class="slide level2">

<h3 id="case-study-ii">case study II</h3>
<p><strong>Ohio State University</strong></p>
<ul>
<li><strong>Integration</strong> and <strong>analysis</strong> of Chip-seq data</li>
<li><strong>Modeling</strong> genomic information and <strong>gene regulatory networks</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="case-study-iii">case study III</h3>
<p><strong>Berkeley Phylogenomics Group</strong></p>
<ul>
<li>Graph database for <em>Big Data challenges</em> in <strong>genomics</strong> developed <strong>on top of Bio4j</strong></li>
</ul>
</section>
<section class="slide level2">

<h3 id="community">community</h3>
<ul>
<li><strong><a href="http://twitter.com/bio4j">@bio4j</a></strong> twitter</li>
<li><strong><a href="https://github.com/bio4j">bio4j</a></strong> github org</li>
<li><strong><a href="http://groups.google.com/group/bio4j-user">bio4j-user</a></strong> google group</li>
<li><strong><a href="http://www.linkedin.com/groups/Bio4j-3890937">bio4j</a></strong> linkedin</li>
</ul>
</section>
<section class="slide level2">

<p><br/> <br/></p>
<!-- How Bio4j -->
<h3 id="whos-doing-bio4j"><em>who</em>’s doing Bio4j?</h3>
<p>research group, team</p>
</section>
<section class="slide level2">

<h3 id="oh-no-sequences">oh no sequences!</h3>
<p><strong><a href="http://era7bioinformatics.com">Era7 bioinformatics</a></strong> R&amp;D group</p>
<ul>
<li><strong>web</strong> -&gt; <a href="http://ohnosequences.com">ohnosequences.com</a></li>
<li><strong>Github</strong> -&gt; <a href="https://github.com/ohnosequences">ohnosequences</a></li>
</ul>
</section>
<section class="slide level2">

<h3 id="team">team</h3>
<ul>
<li><strong><a href="http://ohnosequences.com/ppareja">Pablo Pareja</a></strong> <br /> project leader &amp; main dev</li>
<li><strong><a href="http://ohnosequences.com/eparejatobes">Eduardo Pareja-Tobes</a></strong><br /> technology &amp; architecture</li>
<li><strong><a href="http://ohnosequences.com/rtobes">Raquel Tobes</a></strong> <br /> bio data integration</li>
</ul>
</section>
<section class="slide level2">

<h3 id="team-1">team</h3>
<ul>
<li><strong><a href="http://ohnosequences.com/aalekhin">Alexey Alekhin</a></strong> <br /> Statika, release process, dev</li>
<li><strong><a href="http://ohnosequences.com/mmanrique">Marina Manrique</a></strong> <br /> bio data integration</li>
<li><strong><a href="http://ohnosequences.com/ekovach">Evdokim Kovach</a></strong><br /> dev</li>
</ul>
</section>
<section class="slide level2">

</section>
<!-- {{{{ dzslides core
#
#
#     __  __  __       .  __   ___  __
#    |  \  / /__` |    | |  \ |__  /__`
#    |__/ /_ .__/ |___ | |__/ |___ .__/ core :€
#
#
# The following block of code is not supposed to be edited.
# But if you want to change the behavior of these slides,
# feel free to hack it!
#
-->

<div id="progress-bar"></div>

<!-- Default Style -->
<style>
  * { margin: 0; padding: 0; -moz-box-sizing: border-box; -webkit-box-sizing: border-box; box-sizing: border-box; }
  details { display: none; }
  body {
    width: 800px; height: 600px;
    margin-left: -400px; margin-top: -300px;
    position: absolute; top: 50%; left: 50%;
    overflow: hidden;
  }
  section {
    position: absolute;
    pointer-events: none;
    width: 100%; height: 100%;
  }
  section[aria-selected] { pointer-events: auto; }
  html { overflow: hidden; }
  body { display: none; }
  body.loaded { display: block; }
  .incremental {visibility: hidden; }
  .incremental[active] {visibility: visible; }
  #progress-bar{
    bottom: 0;
    position: absolute;
    -moz-transition: width 400ms linear 0s;
    -webkit-transition: width 400ms linear 0s;
    -ms-transition: width 400ms linear 0s;
    transition: width 400ms linear 0s;
  }
  figure {
    width: 100%;
    height: 100%;
  }
  figure > * {
    position: absolute;
  }
  figure > img, figure > video {
    width: 100%; height: 100%;
  }
</style>

<script>
  var Dz = {
    remoteWindows: [],
    idx: -1,
    step: 0,
    slides: null,
    progressBar : null,
    params: {
      autoplay: "1"
    }
  };

  Dz.init = function() {
    document.body.className = "loaded";
    this.slides = $$("body > section");
    this.progressBar = $("#progress-bar");
    this.setupParams();
    this.onhashchange();
    this.setupTouchEvents();
    this.onresize();
  }
  
  Dz.setupParams = function() {
    var p = window.location.search.substr(1).split('&');
    p.forEach(function(e, i, a) {
      var keyVal = e.split('=');
      Dz.params[keyVal[0]] = decodeURIComponent(keyVal[1]);
    });
  // Specific params handling
    if (!+this.params.autoplay)
      $$.forEach($$("video"), function(v){ v.controls = true });
  }

  Dz.onkeydown = function(aEvent) {
    // Don't intercept keyboard shortcuts
    if (aEvent.altKey
      || aEvent.ctrlKey
      || aEvent.metaKey
      || aEvent.shiftKey) {
      return;
    }
    if ( aEvent.keyCode == 37 // left arrow
      || aEvent.keyCode == 38 // up arrow
      || aEvent.keyCode == 33 // page up
    ) {
      aEvent.preventDefault();
      this.back();
    }
    if ( aEvent.keyCode == 39 // right arrow
      || aEvent.keyCode == 40 // down arrow
      || aEvent.keyCode == 34 // page down
    ) {
      aEvent.preventDefault();
      this.forward();
    }
    if (aEvent.keyCode == 35) { // end
      aEvent.preventDefault();
      this.goEnd();
    }
    if (aEvent.keyCode == 36) { // home
      aEvent.preventDefault();
      this.goStart();
    }
    if (aEvent.keyCode == 32) { // space
      aEvent.preventDefault();
      this.toggleContent();
    }
    if (aEvent.keyCode == 70) { // f
      aEvent.preventDefault();
      this.goFullscreen();
    }
  }

  /* Touch Events */

  Dz.setupTouchEvents = function() {
    var orgX, newX;
    var tracking = false;

    var db = document.body;
    db.addEventListener("touchstart", start.bind(this), false);
    db.addEventListener("touchmove", move.bind(this), false);

    function start(aEvent) {
      aEvent.preventDefault();
      tracking = true;
      orgX = aEvent.changedTouches[0].pageX;
    }

    function move(aEvent) {
      if (!tracking) return;
      newX = aEvent.changedTouches[0].pageX;
      if (orgX - newX > 100) {
        tracking = false;
        this.forward();
      } else {
        if (orgX - newX < -100) {
          tracking = false;
          this.back();
        }
      }
    }
  }

  /* Adapt the size of the slides to the window */

  Dz.onresize = function() {
    var db = document.body;
    var sx = db.clientWidth / window.innerWidth;
    var sy = db.clientHeight / window.innerHeight;
    var transform = "scale(" + (1/Math.max(sx, sy)) + ")";

    db.style.MozTransform = transform;
    db.style.WebkitTransform = transform;
    db.style.OTransform = transform;
    db.style.msTransform = transform;
    db.style.transform = transform;
  }


  Dz.getDetails = function(aIdx) {
    var s = $("section:nth-of-type(" + aIdx + ")");
    var d = s.$("details");
    return d ? d.innerHTML : "";
  }

  Dz.onmessage = function(aEvent) {
    var argv = aEvent.data.split(" "), argc = argv.length;
    argv.forEach(function(e, i, a) { a[i] = decodeURIComponent(e) });
    var win = aEvent.source;
    if (argv[0] === "REGISTER" && argc === 1) {
      this.remoteWindows.push(win);
      this.postMsg(win, "REGISTERED", document.title, this.slides.length);
      this.postMsg(win, "CURSOR", this.idx + "." + this.step);
      return;
    }
    if (argv[0] === "BACK" && argc === 1)
      this.back();
    if (argv[0] === "FORWARD" && argc === 1)
      this.forward();
    if (argv[0] === "START" && argc === 1)
      this.goStart();
    if (argv[0] === "END" && argc === 1)
      this.goEnd();
    if (argv[0] === "TOGGLE_CONTENT" && argc === 1)
      this.toggleContent();
    if (argv[0] === "SET_CURSOR" && argc === 2)
      window.location.hash = "#" + argv[1];
    if (argv[0] === "GET_CURSOR" && argc === 1)
      this.postMsg(win, "CURSOR", this.idx + "." + this.step);
    if (argv[0] === "GET_NOTES" && argc === 1)
      this.postMsg(win, "NOTES", this.getDetails(this.idx));
  }

  Dz.toggleContent = function() {
    // If a Video is present in this new slide, play it.
    // If a Video is present in the previous slide, stop it.
    var s = $("section[aria-selected]");
    if (s) {
      var video = s.$("video");
      if (video) {
        if (video.ended || video.paused) {
          video.play();
        } else {
          video.pause();
        }
      }
    }
  }

  Dz.setCursor = function(aIdx, aStep) {
    // If the user change the slide number in the URL bar, jump
    // to this slide.
    aStep = (aStep != 0 && typeof aStep !== "undefined") ? "." + aStep : ".0";
    window.location.hash = "#" + aIdx + aStep;
  }

  Dz.onhashchange = function() {
    var cursor = window.location.hash.split("#"),
        newidx = 1,
        newstep = 0;
    if (cursor.length == 2) {
      newidx = ~~cursor[1].split(".")[0];
      newstep = ~~cursor[1].split(".")[1];
      if (newstep > Dz.slides[newidx - 1].$$('.incremental > *').length) {
        newstep = 0;
        newidx++;
      }
    }
    this.setProgress(newidx, newstep);
    if (newidx != this.idx) {
      this.setSlide(newidx);
    }
    if (newstep != this.step) {
      this.setIncremental(newstep);
    }
    for (var i = 0; i < this.remoteWindows.length; i++) {
      this.postMsg(this.remoteWindows[i], "CURSOR", this.idx + "." + this.step);
    }
  }

  Dz.back = function() {
    if (this.idx == 1 && this.step == 0) {
      return;
    }
    if (this.step == 0) {
      this.setCursor(this.idx - 1,
                     this.slides[this.idx - 2].$$('.incremental > *').length);
    } else {
      this.setCursor(this.idx, this.step - 1);
    }
  }

  Dz.forward = function() {
    if (this.idx >= this.slides.length &&
        this.step >= this.slides[this.idx - 1].$$('.incremental > *').length) {
        return;
    }
    if (this.step >= this.slides[this.idx - 1].$$('.incremental > *').length) {
      this.setCursor(this.idx + 1, 0);
    } else {
      this.setCursor(this.idx, this.step + 1);
    }
  }

  Dz.goStart = function() {
    this.setCursor(1, 0);
  }

  Dz.goEnd = function() {
    var lastIdx = this.slides.length;
    var lastStep = this.slides[lastIdx - 1].$$('.incremental > *').length;
    this.setCursor(lastIdx, lastStep);
  }

  Dz.setSlide = function(aIdx) {
    this.idx = aIdx;
    var old = $("section[aria-selected]");
    var next = $("section:nth-of-type("+ this.idx +")");
    if (old) {
      old.removeAttribute("aria-selected");
      var video = old.$("video");
      if (video) {
        video.pause();
      }
    }
    if (next) {
      next.setAttribute("aria-selected", "true");
      var video = next.$("video");
      if (video && !!+this.params.autoplay) {
        video.play();
      }
    } else {
      // That should not happen
      this.idx = -1;
      // console.warn("Slide doesn't exist.");
    }
  }

  Dz.setIncremental = function(aStep) {
    this.step = aStep;
    var old = this.slides[this.idx - 1].$('.incremental > *[aria-selected]');
    if (old) {
      old.removeAttribute('aria-selected');
    }
    var incrementals = $$('.incremental');
    if (this.step <= 0) {
      $$.forEach(incrementals, function(aNode) {
        aNode.removeAttribute('active');
      });
      return;
    }
    var next = this.slides[this.idx - 1].$$('.incremental > *')[this.step - 1];
    if (next) {
      next.setAttribute('aria-selected', true);
      next.parentNode.setAttribute('active', true);
      var found = false;
      $$.forEach(incrementals, function(aNode) {
        if (aNode != next.parentNode)
          if (found)
            aNode.removeAttribute('active');
          else
            aNode.setAttribute('active', true);
        else
          found = true;
      });
    } else {
      setCursor(this.idx, 0);
    }
    return next;
  }

  Dz.goFullscreen = function() {
    var html = $('html'),
        requestFullscreen = html.requestFullscreen || html.requestFullScreen || html.mozRequestFullScreen || html.webkitRequestFullScreen;
    if (requestFullscreen) {
      requestFullscreen.apply(html);
    }
  }
  
  Dz.setProgress = function(aIdx, aStep) {
    var slide = $("section:nth-of-type("+ aIdx +")");
    if (!slide)
      return;
    var steps = slide.$$('.incremental > *').length + 1,
        slideSize = 100 / (this.slides.length - 1),
        stepSize = slideSize / steps;
    this.progressBar.style.width = ((aIdx - 1) * slideSize + aStep * stepSize) + '%';
  }
  
  Dz.postMsg = function(aWin, aMsg) { // [arg0, [arg1...]]
    aMsg = [aMsg];
    for (var i = 2; i < arguments.length; i++)
      aMsg.push(encodeURIComponent(arguments[i]));
    aWin.postMessage(aMsg.join(" "), "*");
  }
  
  function init() {
    Dz.init();
    window.onkeydown = Dz.onkeydown.bind(Dz);
    window.onresize = Dz.onresize.bind(Dz);
    window.onhashchange = Dz.onhashchange.bind(Dz);
    window.onmessage = Dz.onmessage.bind(Dz);
  }

  window.onload = init;
</script>


<script> // Helpers
  if (!Function.prototype.bind) {
    Function.prototype.bind = function (oThis) {

      // closest thing possible to the ECMAScript 5 internal IsCallable
      // function 
      if (typeof this !== "function")
      throw new TypeError(
        "Function.prototype.bind - what is trying to be fBound is not callable"
      );

      var aArgs = Array.prototype.slice.call(arguments, 1),
          fToBind = this,
          fNOP = function () {},
          fBound = function () {
            return fToBind.apply( this instanceof fNOP ? this : oThis || window,
                   aArgs.concat(Array.prototype.slice.call(arguments)));
          };

      fNOP.prototype = this.prototype;
      fBound.prototype = new fNOP();

      return fBound;
    };
  }

  var $ = (HTMLElement.prototype.$ = function(aQuery) {
    return this.querySelector(aQuery);
  }).bind(document);

  var $$ = (HTMLElement.prototype.$$ = function(aQuery) {
    return this.querySelectorAll(aQuery);
  }).bind(document);

  $$.forEach = function(nodeList, fun) {
    Array.prototype.forEach.call(nodeList, fun);
  }

</script>
<!-- vim: set fdm=marker: }}} -->
</body>
</html>
