
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Automatic annotation of E. coli LB226692 genome - oh no sequences!</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="And this morning we’ve finished the automatic annotation of the other isolate from the outbreak whose sequence is also available. The isolate &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ohnosequences.com/blog/2011/06/08/automatic-annotation-of-e-coli-lb226692-genome">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="oh no sequences!" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-4']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1>
    <a href="/">
      <span class="title_oh">oh</span><span class="title_no">no</span><span class="title_sequences">sequences</span><span class="title_bang">!</span>
    </a>
  </h1>
  <h2>era7 bioinformatics R&D group</h2>

</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:ohnosequences.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/publications">publications</a></li>
  <li><a href="/projects">projects</a></li>
  <li><a href="/people">people</a></li>
  <li><a href="/about">about</a></li>
  <li><a href="/contact">contact</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      <h1 class="entry-title">Automatic annotation of E. coli LB226692 genome</h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-06-08T12:35:22+02:00" pubdate data-updated="true">Jun 8<span>th</span>, 2011</time>
        
      </p>
    
  </header>


<div class="entry-content"><p>And this morning we’ve finished the automatic annotation of the other isolate from the outbreak whose sequence is also available. The isolate <code>LB226692</code>.</p>

<p>Life Tech and University of Muenster have sequenced the genome and done a hybrid mapping assembly getting finally 364 contigs. More details on the assembly method <a href="https://github.com/ehec-outbreak-crowdsourced/BGI-data-analysis/wiki/Assemblies">here</a></p>

<p>We used<a href="http://www.slideshare.net/marina_manrique/bg7-a-new-system-for-bacterial-genome-annotation-designed-for-ngs-data">BG7 system</a> to annotate the genome.</p>

<p>In this case we’ve selected as reference protein the same protein set we used to annotate the other E. coli isolate (the <code>TY-2482</code>). This protein set has 137,063 proteins and includes:</p>

<ul>
  <li>The representative Uniprot proteins corresponding to all Uniref90 clusters for all Escherichia coli proteins</li>
  <li>All Uniprot proteins from organisms including in their name the terms “EHEC” or “EAEC”</li>
  <li>All Uniprot proteins from bacteria that have in any Uniprot field the term “toxin”</li>
  <li>All Uniprot proteins from bacteria that have in any Uniprot field  “hemolysin”</li>
  <li>All the proteins from Salmonella typhi, Yersinia pestis and Shigella dysenteriae</li>
</ul>

<h4 id="results">Results</h4>

<p>We’ve detected <strong>6,302</strong> genes</p>

<ul>
  <li>6,132 protein encoding genes</li>
  <li>170 RNA genes</li>
</ul>

<p>4,504 out of the 6,132 (73.45%) protein encoding genes have canonical start and stop codon and haven´t either frame-shifts or intragenic stop codons.</p>

<p>1,125 out of the 6,132 (18.34%) protein encoding genes have some frameshifts or intragenic stop codon in their sequences, probably caused by inherent technology errors.</p>

<p>You can get the results of the annotation here <a href="https://github.com/ehec-outbreak-crowdsourced/BGI-data-analysis/tree/master/strains/LB226692/annotations/era7bioinformatics">https://github.com/ehec-outbreak-crowdsourced/BGI-data-analysis/tree/master/strains/LB226692/annotations/era7bioinformatics</a></p>
</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Raquel Tobes</span></span>

      








  


<time datetime="2011-06-08T12:35:22+02:00" pubdate data-updated="true">Jun 8<span>th</span>, 2011</time>
      

<span class="categories">
  
    <a class='category' href='/blog/categories/ngs/'>ngs</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://ohnosequences.com/blog/2011/06/08/automatic-annotation-of-e-coli-lb226692-genome/" data-via="ohnoseq" data-counturl="http://ohnosequences.com/blog/2011/06/08/automatic-annotation-of-e-coli-lb226692-genome/" >Tweet</a>
  
  
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/blog/2011/06/08/automatic-annotation-of-the-second-bgi-assembly-of-e-coli-ty-2482-genome/" title="Previous Post: Automatic annotation of the second BGI assembly of E. coli TY-2482 genome">&laquo; Automatic annotation of the second BGI assembly of E. coli TY-2482 genome</a>
      
      
        <a class="basic-alignment right" href="/blog/2011/06/08/are-there-plasmids-in-the-e-coli-ty-2482-genome/" title="Next Post: Are there plasmids in the E. coli TY-2482 genome?">Are there plasmids in the E. coli TY-2482 genome? &raquo;</a>
      
    </p>
  </footer>
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2013/12/03/intercrossing-cloud-and-ngs-course/">INTERCROSSING cloud and NGS course</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/21/a-short-report-about-the-conference-exploring-human-host-microbiome-interactions-in-health-and-disease-at-wellcome-trust-hinxton-on-may-8-10-2012/">A short report about the Conference Exploring Human Host-Microbiome Interactions in Health and Disease at Wellcome Trust, Hinxton on May 8 - 10, 2012 </a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/23/neo4j-server-and-aws-become-good-friends/">Neo4j Server and AWS become good friends!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/10/18/playing-with-microsatellites-simple-sequence-repeats-java-and-neo4j/">Playing with microsatellites (Simple Sequence Repeats), Java, and Neo4j</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/ohnosequences">@ohnosequences</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'ohnosequences',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2014 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
