
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>oh no sequences!</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="Hi! The first version of Bio4j including RefSeq data is already available.
More information here ">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ohnosequences.com/blog/page/3">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="oh no sequences!" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-4']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1>
    <a href="/">
      <span class="title_oh">oh</span><span class="title_no">no</span><span class="title_sequences">sequences</span><span class="title_bang">!</span>
    </a>
  </h1>
  <h2>era7 bioinformatics R&D group</h2>

</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:ohnosequences.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/publications">publications</a></li>
  <li><a href="/projects">projects</a></li>
  <li><a href="/people">people</a></li>
  <li><a href="/about">about</a></li>
  <li><a href="/contact">contact</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div class="blog-index">
  
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/05/30/bio4j-includes-refseq-data-now/">Bio4j includes RefSeq data now !</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-05-30T17:26:31+02:00" pubdate data-updated="true">May 30<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi!</p>

<p>The first version of Bio4j including RefSeq data is already available.
More information <a href="http://blog.bio4j.com/2011/05/bio4j-includes-refseq-data-now/">here</a></p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/04/11/bio4j-go-tools-goes-web/">Bio4j GO Tools goes web</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-04-11T19:11:11+02:00" pubdate data-updated="true">Apr 11<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Hi!</p>

<p>From now on Bio4j GO Tools will be available <a href="http://blog.bio4j.com/2011/04/bio4j-go-tools-goes-web/">here</a>.</p>

<p>Remember to check the <a href="blog.bio4j.com">Bio4j Blog</a> for updates.</p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/03/29/playing-with-gephi-bio4j-and-go/">Playing with Gephi, Bio4j and Go</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-03-29T17:27:17+02:00" pubdate data-updated="true">Mar 29<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>It had already been some time without having some fun with <a href="http://www.gephi.org"><strong>Gephi</strong></a> so today I told myself: why not trying visualizing the whole Gene Ontology and seeing what happens?</p>

<p>First of all I had to generate the corresponding file in <a href="http://gexf.net/format/">gexf format</a> containing all the terms and relationships belonging to the ontology.
For that I did a small program (<a href="https://github.com/bio4j/Bio4jTools/blob/master/src/com/era7/bioinfo/bio4j/tools/GenerateGexfGo.java"><strong>GenerateGexfGo.java</strong></a>) which uses <a href="http://www.bio4j.com"><strong>Bio4j</strong></a> for terms/relationships info retrieval and a couple of XML Gexf wrapper classes from the github project <a href="https://github.com/pablopareja/BioinfoXML">BioinfoXML</a>.</p>

<p>Once I had my gexf file I tried opening it (~17 MB) with gephi in my laptop with no success, (<em>gephi froze forever when trying to import the file</em>). 
Then, after a quick search on google I figured out that the amount of memory used by Gephi was really easy to change, (<em>just open the file ‘etc/gephi07beta.conf’ and change the -Xmx value</em>).</p>

<p>With my file already imported, first I applied the algorithm <a href="http://gephi.org/2010/openord-new-layout-plugin-the-fastest-algorithm-so-far/"><strong>OpenOrd</strong></a> (which is the best one for large graphs) and then once it had an acceptable distribution I finally applied some iterations of the algorithm Fruchterman Reingold for a better visualization.
And this is what I got:</p>

<p><img src="/images/wholeGo.svg" /></p>

<p>Colors correspondance:	</p>

<ul>
  <li><strong>Green</strong>: Cellular component</li>
  <li><strong>Blue</strong>: Molecular function</li>
  <li><strong>Orange</strong>: Biological process</li>
</ul>

<p><strong>UPDATE:</strong> zoomable independent ontology visualizations using gephi <a href="http://gephi.org/plugins/seadragon/">SeaDragon plugin</a>.</p>

<p><strong>Molecular function</strong></p>

<p><img src="/images/molecularFunctionScreenShot-300x281.jpg" /></p>

<p><strong>Cellular component</strong></p>

<p><img src="/images/cellularComponentScreenShot-300x280.jpg" /></p>

<p><strong>Biological process</strong></p>

<p><img src="/images/biologicalProcessScreenShot-300x275.jpg" /></p>

<p><a href="https://s3-eu-west-1.amazonaws.com/pablo-tests/wholeGo.gexf">Here</a> you can download the gexf file in case you want to experiment a bit with it.</p>

</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/03/25/using-bio4j-go-tools-to-get-the-go-annotations-of-a-set-of-uniprot-proteins-2/">Using Bio4j Go Tools to get the GO annotations of a set of Uniprot proteins</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-03-25T18:29:11+01:00" pubdate data-updated="true">Mar 25<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>We have recently launched a first version of Bio4j Go Tools: an <a href="https://github.com/bio4j/Bio4jGoTools/blob/master/releases/Bio4jGoTools.air">AIR application </a>that allows the user to perform different kinds of GO Analysis using <a href="bio4j.com">Bio4j </a>as back-end.
More info about the tool in <a href="http://blog.bio4j.com/?p=9">this post</a> of <a href="http://blog.bio4j.com/">Bio4j blog</a></p>

<p><img src="/images/Bio4jJPG.jpg" /></p>

<p>So far you can do the following:</p>

<ol>
  <li>Get the GO annotations of a set of proteins. All the GO annotations belonging to the three different GO ontologies (Molecular function, Biological process y Cellular component) are provided separately</li>
  <li>“Perform” a GO Slim analysis</li>
</ol>

<p>Using this tool for 1) is pretty easy. You just have to enter in the text field the Uniprot IDs (protein accessions) of the set of proteins of interest and select which kind of separator you are using (enter, tab, whitespace…). Then click on “Get results”, select the location where the result file will be downloaded and wait until the “File downloaded!” message appears.</p>

<p>New charts utilities are coming soon. By now, you can import the XML result file into an Excel file and create the charts there. The following ones are an example of the charts you can make. They represent the GO annotations of all the <em>Helicobacter pylori</em> proteins (11,934 proteins), only those GO terms that appear 20 times or more in the set of proteins are depicted.</p>

<p><img src="/images/HPylori_BP2.jpg" /></p>

<p><img src="/images/HPylori_CC.jpg" /></p>

<p><img src="/images/HPylori_MF.jpg)" /></p>

<p>How to use this tool to perform GOSlim analysis in upcoming posts…</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/03/23/bio4j-a-graph-based-db-containing-most-of-data-available-in-uniprot-uniref-and-gene-ontology/">Bio4j: a graph based DB containing most of data available in Uniprot, Uniref and Gene Ontology</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-03-23T14:29:56+01:00" pubdate data-updated="true">Mar 23<span>rd</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>We have recently launched Bio4j, a graph based database that includes most of the data available in Uniprot, Uniref clusters and Gene Ontology. Quoting from the Bio4j wiki:</p>

<p> 
&gt; Bio4j provides a completely new and powerful framework for protein related information querying and management. Since it relies on a high-performance graph engine (Neo4j), data is stored in a way that semantically represents its own structure. On the contrary, traditional relational databases must flatten the data they represent into tables, creating “artificial” ids in order to connect the different tuples; which can in some cases eventually lead to domain models that have almost nothing to do with the actual structure of data.</p>

<p>Thanks to its graph structure queries that could be time consuming or even inpossible in relational DBs can be performed easily in Bio4j.</p>

<p>An example of this kind of queries would be the following:</p>

<p>Suppose you are working in the membrane proteins of a non model organism that unfortunately have very poor annotations. Searching for proteins of your interest in Uniprot by your organism name or taxonomy only results in a few unrewieved entries with few more than the sequence. Next step could probably be to use the Uniprot advance search tool to get the proteins similar to your proteins in a 50, 90 or 100% (the Uniref clusters) and then look protein by protein which of them have annotations that could indicate that they are membrane proteins and select those that 1) have good enough annotaions for your purpose and 2) are related enough to your protein to be able to infer the functions of the well-annotated proteins to your protein. This would be a manual and tedious process time consuming and error prone.</p>

<p>All this process of getting all the proteins of your organism, then getting the high similar ones and then getting only those which are in the membrane and have quality annotations could be perfomed programmatically in a fairly simple graph traversal. All you would have to specify would be:</p>

<ul>
  <li>The level of needed similarity between the proteins of your organism and the well-characterized ones</li>
  <li>The criteria to consider a protein has good enough annotations: if the entry is reviewed, if it’s got lots of cross references, of functional data (Interpro, GO…)</li>
  <li>The criteria to consider a protein is in the membrane, for example that it’s got a GO ID indicating that is a membrane protein, that its subcellular location is the membrane… both…</li>
</ul>

<p>More information about Bio4j in its <a href="http://www.bio4j.com">website</a>, the <a href="http://blog.bio4j.com/">blog</a>, the <a href="http://wiki.bio4j.com/">wiki </a> and the <a href="http://twitter.com/#!/bio4j">twitter profile</a>.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/02/25/using-uniprot-advanced-search/">Using Uniprot Advanced search</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-02-25T16:06:18+01:00" pubdate data-updated="true">Feb 25<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>This short post shows how to use Uniprot Advanced search to retrieve proteins of interest. In this Biostar <a href="http://biostar.stackexchange.com/questions/5862/how-to-retrieve-human-proteins-sequence-containing-a-given-domain">question</a> they want to retrieve human proteins with the domain “FYVE”</p>

<p>First click on <em>Advanced search</em> in Uniprot (www.uniprot.org) website</p>

<p><img src="/images/1.jpg" /></p>

<p>Then select <em>Domain</em> in the field box and type <em>FYVE</em></p>

<p><img src="/images/2.jpg" /></p>

<p>Click then <em>Add &amp; Search</em> and select now <em>Organism</em> in the Field box and type <em>Homo sapiens</em>.</p>

<p><img src="/images/3.jpg" /></p>

<p>Clicking again on <em>Add &amp; Search</em> you get your set of proteins. And if you click on <em>Download</em> on the right you can download the data in several formats.</p>

<p>Similar searches (and more complex ones) can be done combining the logical operators and the available fields.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/02/23/bacterial-diversity-identification-with-normalized-libraries/">Bacterial diversity identification with normalized libraries?</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-02-23T14:07:40+01:00" pubdate data-updated="true">Feb 23<span>rd</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>Normally the bacterial identification experiments with NGS are based on analyzing 16S rRNA amplicons. Typically with 454 technology. In this kind of experiments the diversity of the bacterial populations found in the samples are infered from the similarity of the reads with already known 16s rRNA sequences. And the abundance of detected species could be measured with the abundance of detected reads of such specie in the sample.</p>

<p>This kind of approach where the abundance of a certain thing is meassured with the abundance of annotated sequences with that thing is also applied in RNA-seq experiments (where the gene expression is such thing) with no normalized libraries.</p>

<p>If the goal of the transcriptome experiment is not to quantify gene expression but detect as many different transcripts as possible then the library is normalized.</p>

<p>I wonder if in the bacterial identificacion experiments using 16s amplicons this approach could also be used in the cases where you want to get the maximum number of species present in your sample.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/02/14/still-playing-with-gephi-and-go-i/">Still playing with Gephi and GO (I)</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-02-14T17:43:07+01:00" pubdate data-updated="true">Feb 14<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>I was wondering the other day how would the whole <a href="http://www.geneontology.org/">Gene Ontology</a> look like in <a href="http://gephi.org/">Gephi</a>. After realizing that it exporting the ontology to a gexf file could be pretty easy I couldn’t resist to give a try.
Soon enough I realized that representing the three sub-ontologies of <a href="http://www.geneontology.org/">GO</a> (<em>biological process, cellular component and molecular function</em>) at the same time was kind of too much. So in the end I decided to start with the smallest one (<em>cellular component</em>), and this is what I got.</p>

<p>Click <a href="/images/cellularComponent.pdf"><strong>here</strong></a> to open the pdf in a new window.</p>
</div>
  
  


    </article>
  
  
    <article>
      
  <header>
    
      <h1 class="entry-title"><a href="/blog/2011/02/07/using-gephi-with-gene-ontology-graphs/">Using Gephi with gene ontology graphs</a></h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-02-07T18:49:01+01:00" pubdate data-updated="true">Feb 7<span>th</span>, 2011</time>
        
      </p>
    
  </header>


  <div class="entry-content"><p>It’s already been a couple of weeks since we started integrating <a href="http://gephi.org/"><strong>Gephi</strong></a> in some of our services and projects and I must say I’m quite impressed with the results so far.</p>

<p>One of the first uses we found for it was representing gene ontology protein annotations and Go Slim analysis.</p>

<p>Here you can see a graph representation of a protein set gene ontology annotations. Different colors have been applied to each ontology (<em>molecular function, biological process and cellular component</em>). Besides, nodes size is proportional to protein annotations number.</p>

<p>Click <a href="/images/GoSlimBlog.svg"><strong>here</strong></a> to open the svg in a new window.</p>

</div>
  
  


    </article>
  
  <div class="pagination">
    
    <a href="/blog/archives">Blog Archives</a>
    
    <a class="next" href="/blog/page/2/">Newer &rarr;</a>
    
  </div>
</div>
<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2013/12/03/intercrossing-cloud-and-ngs-course/">INTERCROSSING cloud and NGS course</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/21/a-short-report-about-the-conference-exploring-human-host-microbiome-interactions-in-health-and-disease-at-wellcome-trust-hinxton-on-may-8-10-2012/">A short report about the Conference Exploring Human Host-Microbiome Interactions in Health and Disease at Wellcome Trust, Hinxton on May 8 - 10, 2012 </a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/23/neo4j-server-and-aws-become-good-friends/">Neo4j Server and AWS become good friends!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/10/18/playing-with-microsatellites-simple-sequence-repeats-java-and-neo4j/">Playing with microsatellites (Simple Sequence Repeats), Java, and Neo4j</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating&#8230;</li>
  </ul>
  
  <a href="https://github.com/ohnosequences">@ohnosequences</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'ohnosequences',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


  
</aside>
    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2014 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
