
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Playing with Gephi, Bio4j and Go - oh no sequences!</title>
  <meta name="author" content="oh no sequences!">

  
  <meta name="description" content="It had already been some time without having some fun with Gephi so today I told myself: why not trying visualizing the whole Gene Ontology and &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ohnosequences.com/blog/2011/03/29/playing-with-gephi-bio4j-and-go">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="oh no sequences!" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href='http://fonts.googleapis.com/css?family=Open+Sans:600italic,300italic,400,400italic,600,300' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bree+Serif' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Bitter:400,400italic' rel='stylesheet' type='text/css'>
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  showMathMenu: false,
  TeX: {
    extensions: ["AMSmath.js", "AMSsymbols.js"]
  }, 
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", webFont: "TeX", availableFonts: [] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>


  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-12161444-4']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1>
    <a href="/">
      <span class="title_oh">oh</span><span class="title_no">no</span><span class="title_sequences">sequences</span><span class="title_bang">!</span>
    </a>
  </h1>
  <h2>era7 bioinformatics R&D group</h2>

</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:ohnosequences.com" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/blog">blog</a></li>
  <li><a href="/publications">publications</a></li>
  <li><a href="/projects">projects</a></li>
  <li><a href="/people">people</a></li>
  <li><a href="/about">about</a></li>
  <li><a href="/contact">contact</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      <h1 class="entry-title">Playing with Gephi, Bio4j and Go</h1>
    
    
      <p class="meta">
        








  


<time datetime="2011-03-29T17:27:17+02:00" pubdate data-updated="true">Mar 29<span>th</span>, 2011</time>
        
      </p>
    
  </header>


<div class="entry-content"><p>It had already been some time without having some fun with <a href="http://www.gephi.org"><strong>Gephi</strong></a> so today I told myself: why not trying visualizing the whole Gene Ontology and seeing what happens?</p>

<p>First of all I had to generate the corresponding file in <a href="http://gexf.net/format/">gexf format</a> containing all the terms and relationships belonging to the ontology.
For that I did a small program (<a href="https://github.com/bio4j/Bio4jTools/blob/master/src/com/era7/bioinfo/bio4j/tools/GenerateGexfGo.java"><strong>GenerateGexfGo.java</strong></a>) which uses <a href="http://www.bio4j.com"><strong>Bio4j</strong></a> for terms/relationships info retrieval and a couple of XML Gexf wrapper classes from the github project <a href="https://github.com/pablopareja/BioinfoXML">BioinfoXML</a>.</p>

<p>Once I had my gexf file I tried opening it (~17 MB) with gephi in my laptop with no success, (<em>gephi froze forever when trying to import the file</em>). 
Then, after a quick search on google I figured out that the amount of memory used by Gephi was really easy to change, (<em>just open the file ‘etc/gephi07beta.conf’ and change the -Xmx value</em>).</p>

<p>With my file already imported, first I applied the algorithm <a href="http://gephi.org/2010/openord-new-layout-plugin-the-fastest-algorithm-so-far/"><strong>OpenOrd</strong></a> (which is the best one for large graphs) and then once it had an acceptable distribution I finally applied some iterations of the algorithm Fruchterman Reingold for a better visualization.
And this is what I got:</p>

<p><img src="/images/wholeGo.svg" /></p>

<p>Colors correspondance:	</p>

<ul>
  <li><strong>Green</strong>: Cellular component</li>
  <li><strong>Blue</strong>: Molecular function</li>
  <li><strong>Orange</strong>: Biological process</li>
</ul>

<p><strong>UPDATE:</strong> zoomable independent ontology visualizations using gephi <a href="http://gephi.org/plugins/seadragon/">SeaDragon plugin</a>.</p>

<p><strong>Molecular function</strong></p>

<p><img src="/images/molecularFunctionScreenShot-300x281.jpg" /></p>

<p><strong>Cellular component</strong></p>

<p><img src="/images/cellularComponentScreenShot-300x280.jpg" /></p>

<p><strong>Biological process</strong></p>

<p><img src="/images/biologicalProcessScreenShot-300x275.jpg" /></p>

<p><a href="https://s3-eu-west-1.amazonaws.com/pablo-tests/wholeGo.gexf">Here</a> you can download the gexf file in case you want to experiment a bit with it.</p>

</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Pablo Pareja</span></span>

      








  


<time datetime="2011-03-29T17:27:17+02:00" pubdate data-updated="true">Mar 29<span>th</span>, 2011</time>
      

<span class="categories">
  
    <a class='category' href='/blog/categories/bio4j/'>bio4j</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://ohnosequences.com/blog/2011/03/29/playing-with-gephi-bio4j-and-go/" data-via="ohnoseq" data-counturl="http://ohnosequences.com/blog/2011/03/29/playing-with-gephi-bio4j-and-go/" >Tweet</a>
  
  
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/blog/2011/03/25/using-bio4j-go-tools-to-get-the-go-annotations-of-a-set-of-uniprot-proteins-2/" title="Previous Post: Using Bio4j Go Tools to get the GO annotations of a set of Uniprot proteins">&laquo; Using Bio4j Go Tools to get the GO annotations of a set of Uniprot proteins</a>
      
      
        <a class="basic-alignment right" href="/blog/2011/04/11/bio4j-go-tools-goes-web/" title="Next Post: Bio4j GO Tools goes web">Bio4j GO Tools goes web &raquo;</a>
      
    </p>
  </footer>
</article>

</div>

<aside class="sidebar">
  
    <section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2015/03/22/bio4j-preprint/">Bio4j preprint</a>
      </li>
    
      <li class="post">
        <a href="/blog/2013/12/03/intercrossing-cloud-and-ngs-course/">INTERCROSSING cloud and NGS course</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/21/a-short-report-about-the-conference-exploring-human-host-microbiome-interactions-in-health-and-disease-at-wellcome-trust-hinxton-on-may-8-10-2012/">A short report about the Conference Exploring Human Host-Microbiome Interactions in Health and Disease at Wellcome Trust, Hinxton on May 8 - 10, 2012 </a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/23/neo4j-server-and-aws-become-good-friends/">Neo4j Server and AWS become good friends!</a>
      </li>
    
      <li class="post">
        <a href="/blog/2011/12/02/gene-ontology-go-graph-visualizations-for-ehec-automatic-annotation-of-bgi-v4-assembly/">Gene Ontology (GO) graph visualizations for EHEC automatic annotation of BGI V4 assembly</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/ohnosequences">@ohnosequences</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'ohnosequences',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2015 - oh no sequences! -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  







  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
